### PR TITLE
Invalidate latest nightly files in CloudFront.

### LINKED ISSUE #29034: SHA256 for Linux binay does not match the provided prebuilt
https://download.servo.org/nightly/linux/servo-latest.tar.gz.sha256 says: `38c60616d71397c143e626eab49112a56cae2a9c7d80ee25d64e1a6a82757e6f`

while the computed SHA256 downloaded from https://servo.org says:
`854b97dafac11953a929b67f9910adb7e6a648721326692ada101de856c0f9b8`.

This already happened before in #28925 with windows builds, and it is happening with Linux builds too.

EDIT: removed unnecessary quotes

### PR DESCRIPTION
**Pull request by mukilan**
After a new nightly build is uploaded to S3, bust the cache in CloudFront Edge servers with the CreateInvalidation AWS API.

For each platform we use the `/nightly/<platform>/servo-latest.<ext>*` pattern to invalidate both package and sha256 files. As part of this change, a new policy has been attached to the "download.servo.org-uploads-from-travis" IAM user
to allow the "cloudfront:CreateInvalidation" action.

Since CloudFront [invalidates every version][1] of the cached file, regardless of the headers used for that version, this change should invalidate the different caches for 'Accept-Encoding' header.

[1]: https://docs.aws.amazon.com/AmazonCloudFront/latest/DeveloperGuide/Invalidation.html#invalidation-specifying-objects

---

- [x] `./mach build -d` does not report any errors
- [x] `./mach test-tidy` does not report any errors
- [x] These changes fix #29034 

- [ ] There are tests for these changes OR
- [x] These changes do not require tests because they update nightly upload process. Changes have been *partially* validated using [manual run ](https://github.com/servo/servo/actions/runs/4012975660/jobs/6892315365)

### CODE REVIEW

File: python/servo/package_commands.py [hunk 1]
```diff
@@ -665,6 +673,21 @@ def upload_to_s3(platform, package, timestamp):
                 package_hash_fileobj, BUCKET, latest_hash_upload_key, ExtraArgs={'ContentType': 'text/plain'}
             )
 
+            # Invalidate previous "latest" nightly files from
+            # CloudFront edge caches
+            cloudfront.create_invalidation(
+                DistributionId=DISTRIBUTION_ID,
+                InvalidationBatch={
+                    'CallerReference': '{}-{}'.format(latest_upload_key, timestamp),
```
Comment (mrobinson) [inline python/servo/package_commands.py:681]:
```suggestion
                    'CallerReference': f'{latest_upload_key}-{timestamp}',
```

File: python/servo/package_commands.py [hunk 2]
```diff
@@ -652,7 +660,7 @@ def upload_to_s3(platform, package, timestamp):
                     sha256_digest.update(data)
             package_hash = sha256_digest.hexdigest()
             package_hash_fileobj = io.BytesIO(package_hash.encode('utf-8'))
-            latest_hash_upload_key = '{}/servo-latest.{}.sha256'.format(nightly_dir, extension)
+            latest_hash_upload_key = '{}.sha256'.format(latest_upload_key)
```
Comment (mrobinson) [inline python/servo/package_commands.py:663]:
```suggestion
            latest_hash_upload_key = f'{latest_upload_key}.sha256'
```

File: python/servo/package_commands.py [hunk 3]
```diff
@@ -665,6 +673,21 @@ def upload_to_s3(platform, package, timestamp):
...
+            # Invalidate previous "latest" nightly files from
+            # CloudFront edge caches
+            cloudfront.create_invalidation(
+                DistributionId=DISTRIBUTION_ID,
+                InvalidationBatch={
+                    'CallerReference': '{}-{}'.format(latest_upload_key, timestamp),
+                    'Paths': {
+                        'Quantity': 1,
+                        'Items': [
+                            '/{}*'.format(latest_upload_key)
```
Comment (mrobinson) [inline python/servo/package_commands.py:685]:
```suggestion
                            f'/{latest_upload_key}*'
```

**Review by mrobinson — APPROVED** [review]
Looks good to me with a few nits.

File: python/servo/package_commands.py (hunk 1 above)
Comment (mukilan) [reply python/servo/package_commands.py]: Addressed in latest push.

File: python/servo/package_commands.py (hunk 2 above)
Comment (mukilan) [reply python/servo/package_commands.py]: Addressed in latest push.

File: python/servo/package_commands.py (hunk 3 above)
Comment (mukilan) [reply python/servo/package_commands.py]: Addressed in latest push.

**Review by delan — APPROVED** [review]

**Comment by mrobinson** [comment]
@bors-servo r+

Just doing this in case things fail. My GitHub account is @mrobinson. :)

**Pull request merged**